### PR TITLE
Add new `value_chain` iterator.

### PR DESCRIPTION
**Pull request by jenstroeger**
See also issue https://github.com/more-itertools/more-itertools/issues/468. This is the non-recursive implementation of the discussed iterator. Happy to add the `collapse` parameter, though.

Also, considering that `value_chain()` is a more general case of `prepend()`, would you like to tail-call:
```python
def prepend(value, iterator):
    return value_chain(value, iterator)
```

### CODE REVIEW

File: more_itertools/more.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
     powerset,
     take,
     unique_everseen,
+    prepend,
```
Comment (jenstroeger) [inline more_itertools/more.py:36]: Hmm. @bbayles, if I remove the `prepend` then Doctest fails, if I add `prepend` then flake8 fails. What now?

File: more_itertools/more.py [hunk 2]
```diff
@@ -3552,3 +3554,39 @@ def nth_permutation(iterable, r, index):
...
+    of :func:`prepend`:
+
+        >>> list(prepend(0, [1, 2, 3]))
+        [0, 1, 2, 3]
+        >>> list(value_chain(0, [1, 2, 3]))
+        [0, 1, 2, 3]
+
+    """
+    for value in args:
+        if isinstance(value, (str, bytes,)):
```
Comment (jenstroeger) [inline more_itertools/more.py:3583]: We should probably add the other scalar types here, too: `int`, `float`, `bool`.

File: more_itertools/more.py (hunk 1 above)
Comment (bbayles) [reply more_itertools/more.py:36]: It shouldn't be imported here; it should be imported in the docstring before [this line](https://github.com/more-itertools/more-itertools/pull/475/files#diff-d7dff201bd69500a573b7e96c52f8e625242d769a04621224112faf6e7bc63fdR3576).

Comment (jenstroeger) [reply more_itertools/more.py]: Duh. Commit https://github.com/more-itertools/more-itertools/pull/475/commits/254133ff846af44c99150b645bd73e4999bd06a3.

File: more_itertools/more.py (hunk 2 above)
Comment (jenstroeger) [reply more_itertools/more.py]: Commit https://github.com/more-itertools/more-itertools/pull/475/commits/0d67cde0543810d8ca75005af2d9c1e94a0c7dd9.

**Comment by jenstroeger** [comment]
Ok so that one failure is a notification from [black](https://github.com/psf/black) which would reformat this:
```python
if isinstance(value, (str, bytes, int, float, bool,)):
```
to this:
```python
if isinstance(
    value,
    (
        str,
        bytes,
        int,
        float,
        bool,
    ),
):
```
This is just plain ugly, IMO, but it’s not my code either so I’m happy to comply.

**Comment by bbayles** [comment]
Go ahead and make the fomatter happy, if you could please.

**Comment by jenstroeger** [comment]
Done 🤓

**Comment by bbayles** [comment]
Merged with some changes:
* Added to API docs
* The `int`, `float`, `bool` code that was added in 0d67cde0543810d8ca75005af2d9c1e94a0c7dd9 weren't tested and were hard to explain - I've left those out.
* Removed references to the `prepend` function - it's a simple recipe from the `itertools` docs that we bring in here for completeness.

Many thanks!

**Pull request merged**